cmake/targets/windows.cmake

```diff
@@ -4,9 +4,3 @@ set(CMAKE_FIND_LIBRARY_SUFFIXES ".dll")
 find_library(ZLIB ZLIB1)
 list(APPEND SUNSHINE_EXTERNAL_LIBRARIES
         Wtsapi32.lib)
-
-#WebUI build
-add_custom_target(web-ui ALL
-        WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
-        COMMENT "Installing NPM Dependencies and Building the Web UI"
-        COMMAND cmd /C "npm install && set \"SUNSHINE_SOURCE_ASSETS_DIR=${NPM_SOURCE_ASSETS_DIR}\" && set \"SUNSHINE_ASSETS_DIR=${NPM_ASSETS_DIR}\" && npm run build")   # cmake-lint: disable=C0301
```

Comment:
Thanks for the PR! Just for informational purposes, this was split to have a windows specific command only because CLion was failing to build the web-ui target, and for some reason it was finding a very old version of npm and node (which I could not even locate on my system). I will have to test your changes locally to make sure it works in CLion as well.

Reply:
I get the following error in CLion with these changes.

```txt
====================[ Build | web-ui | Release ]================================
C:\msys64\mingw64\bin\cmake.exe --build C:\Users\ReenigneArcher\Documents\GitHub\LizardByte\Sunshine\cmake-build-release --target web-ui -- -j 10
[100%] Installing NPM Dependencies and Building the Web UI

up to date, audited 52 packages in 725ms

12 packages are looking for funding
  run `npm fund` for details

2 moderate severity vulnerabilities

To address issues that do not require attention, run:
  npm audit fix

To address all issues, run:
  npm audit fix --force

Run `npm audit` for details.
unknown error
mingw32-make[3]: *** [CMakeFiles\web-ui.dir\build.make:71: CMakeFiles/web-ui] Error 1
mingw32-make[2]: *** [CMakeFiles\Makefile2:240: CMakeFiles/web-ui.dir/all] Error 2
mingw32-make[1]: *** [CMakeFiles\Makefile2:247: CMakeFiles/web-ui.dir/rule] Error 2
mingw32-make: *** [Makefile:202: web-ui] Error 2
```

Reply:
Heh, even my simple changes cause problems! I don't know whether it's the `find_program` or the invocation change that's breaking it here. The output really doesn't reveal much. That audit result was would be fatal for `npm audit` but not `npm install`, so I think that's unrelated. Please could you try it with the two `${NPM}` instances replaced with just `npm`?

Reply:
I agree the audit is probably not the issue.

Removing the find_program and using just `npm` results in:

```txt
====================[ Build | web-ui | Release ]================================
C:\msys64\mingw64\bin\cmake.exe --build C:\Users\ReenigneArcher\Documents\GitHub\LizardByte\Sunshine\cmake-build-release --target web-ui -- -j 10
[100%] Installing NPM Dependencies and Building the Web UI

up to date, audited 52 packages in 718ms

12 packages are looking for funding
  run `npm fund` for details

2 moderate severity vulnerabilities

To address issues that do not require attention, run:
  npm audit fix

To address all issues, run:
  npm audit fix --force

Run `npm audit` for details.
no such file or directory
mingw32-make[3]: *** [CMakeFiles\web-ui.dir\build.make:71: CMakeFiles/web-ui] Error 1
mingw32-make[2]: *** [CMakeFiles\Makefile2:240: CMakeFiles/web-ui.dir/all] Error 2
mingw32-make[1]: *** [CMakeFiles\Makefile2:247: CMakeFiles/web-ui.dir/rule] Error 2
mingw32-make: *** [Makefile:202: web-ui] Error 2
```

changes:
```cmake
#WebUI build
#find_program(NPM npm REQUIRED)
add_custom_target(web-ui ALL
        WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
        COMMENT "Installing NPM Dependencies and Building the Web UI"
        COMMAND npm install
        COMMAND "${CMAKE_COMMAND}" -E env "SUNSHINE_BUILD_HOMEBREW=${NPM_BUILD_HOMEBREW}" "SUNSHINE_SOURCE_ASSETS_DIR=${NPM_SOURCE_ASSETS_DIR}" "SUNSHINE_ASSETS_DIR=${NPM_ASSETS_DIR}" npm run build)  # cmake-lint: disable=C0301
```

Reply:
Adding `--verbose` before the `--` would tell us a lot more too.

Reply:
There is something going on with the way cmake is launching npm. Even running `npm -v` manually in the terminal seems to have some weird looking behavior, almost as if several processes are launching.

I did a simple test.

```cmake
#WebUI build
find_program(NPM npm REQUIRED)

message(STATUS "NPM found at: ${NPM}")

execute_process(
        COMMAND cmd /C "${NPM} -v"
        OUTPUT_VARIABLE NPM_VERSION
        ERROR_VARIABLE NPM_ERROR
        OUTPUT_STRIP_TRAILING_WHITESPACE
        RESULT_VARIABLE result
)

if(result)
    message(FATAL_ERROR "Failed to get NPM version: ${NPM_ERROR}")
else()
    message(STATUS "NPM version: ${NPM_VERSION}")
endif()
```

The above works, but removing the `cmd /c` prefix fails.

And with that, the following works. It might be good to keep the messages for future debugging. You can possibly add a "prefix" command to use if we're on windows?

```cmake
#WebUI build
find_program(NPM npm REQUIRED)

message(STATUS "NPM found at: ${NPM}")

execute_process(
        COMMAND cmd /C "${NPM} -v"
        OUTPUT_VARIABLE NPM_VERSION
        ERROR_VARIABLE NPM_ERROR
        OUTPUT_STRIP_TRAILING_WHITESPACE
        RESULT_VARIABLE result
)

if(result)
    message(FATAL_ERROR "Failed to get NPM version: ${NPM_ERROR}")
else()
    message(STATUS "NPM version: ${NPM_VERSION}")
endif()

add_custom_target(web-ui ALL
        WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
        COMMENT "Installing NPM Dependencies and Building the Web UI"
        COMMAND cmd /C "${NPM} install"
        COMMAND "${CMAKE_COMMAND}" -E env "SUNSHINE_BUILD_HOMEBREW=${NPM_BUILD_HOMEBREW}" "SUNSHINE_SOURCE_ASSETS_DIR=${NPM_SOURCE_ASSETS_DIR}" "SUNSHINE_ASSETS_DIR=${NPM_ASSETS_DIR}" cmd /C "${NPM} run build")  # cmake-lint: disable=C0301
```

I didn't confirm is the env variables are set properly using this method though.

Reply:
It sounds like you're dodging some environmental issue by going via `cmd`. Perhaps you needed to make it `npm.exe` rather than `npm` when trying my suggestion. `--verbose` would definitely help to see what's going on here as it shows what Make is actually doing.

Reply:
using the NPM variable

```txt
Run `npm audit` for details.
cd /d C:\Users\ReenigneArcher\Documents\GitHub\LizardByte\Sunshine && C:\msys64\mingw64\bin\cmake.exe -E env SUNSHINE_BUILD_HOMEBREW= SUNSHINE_SOURCE_ASSETS_DIR=C:/Users/ReenigneArcher/Documents/GitHub/LizardByte/Sunshine/src_assets SUNSHINE_ASSETS_DIR=C:/Users/ReenigneArcher/Documents/GitHub/LizardByte/Sunshine/cmake-build-release C:/msys64/mingw64/bin/npm run build
unknown error
mingw32-make[3]: *** [CMakeFiles\web-ui.dir\build.make:71: CMakeFiles/web-ui] Error 1
mingw32-make[2]: *** [CMakeFiles\Makefile2:240: CMakeFiles/web-ui.dir/all] Error 2
mingw32-make[1]: *** [CMakeFiles\Makefile2:247: CMakeFiles/web-ui.dir/rule] Error 2
mingw32-make: *** [Makefile:202: web-ui] Error 2
mingw32-make[3]: Leaving directory 'C:/Users/ReenigneArcher/Documents/GitHub/LizardByte/Sunshine/cmake-build-release'
mingw32-make[2]: Leaving directory 'C:/Users/ReenigneArcher/Documents/GitHub/LizardByte/Sunshine/cmake-build-release'
mingw32-make[1]: Leaving directory 'C:/Users/ReenigneArcher/Documents/GitHub/LizardByte/Sunshine/cmake-build-release'
```

using `npm` directly
```txt
cd /d C:\Users\ReenigneArcher\Documents\GitHub\LizardByte\Sunshine && C:\msys64\mingw64\bin\cmake.exe -E env SUNSHINE_BUILD_HOMEBREW= SUNSHINE_SOURCE_ASSETS_DIR=C:/Users/ReenigneArcher/Documents/GitHub/LizardByte/Sunshine/src_assets SUNSHINE_ASSETS_DIR=C:/Users/ReenigneArcher/Documents/GitHub/LizardByte/Sunshine/cmake-build-release npm run build
no such file or directory
mingw32-make[3]: *** [CMakeFiles\web-ui.dir\build.make:71: CMakeFiles/web-ui] Error 1
mingw32-make[2]: *** [CMakeFiles\Makefile2:240: CMakeFiles/web-ui.dir/all] Error 2
mingw32-make[1]: *** [CMakeFiles\Makefile2:247: CMakeFiles/web-ui.dir/rule] Error 2
mingw32-make: *** [Makefile:202: web-ui] Error 2
mingw32-make[3]: Leaving directory 'C:/Users/ReenigneArcher/Documents/GitHub/LizardByte/Sunshine/cmake-build-release'
mingw32-make[2]: Leaving directory 'C:/Users/ReenigneArcher/Documents/GitHub/LizardByte/Sunshine/cmake-build-release'
mingw32-make[1]: Leaving directory 'C:/Users/ReenigneArcher/Documents/GitHub/LizardByte/Sunshine/cmake-build-release'
```

Reply:
Okay, thanks. Still not clear what's going on, but I've now used a generator expression to add `cmd /C`.